check recurseForDerivations

tomberek opened the pull request:
Control recursion during eval.
Restructure output to produce a profile.

samrose commented on src/nix-eval-jobs.cc:
```diff
@@ -231,15 +259,25 @@ static void worker(
+                        std::string name(i->name);
+                        if (name.find('.') != std::string::npos || name.find(' ') != std::string::npos) {
+                            name= "\"" + name + "\"";
+                            /* printError("skipping job with illegal name '%s'", name); */
```
maybe remove printErro

tomberek committed:
feat: add depth flag

limit recursion to a default of 1

needs improvement: recursion is only the number of "."

tomberek committed:
feat: output manifest.json compatible objects

only for flakes for now
